### PR TITLE
Adapt to relint 2.0 (issue #4)

### PR DESCRIPTION
**Pull request by mattiase**
This provides locations with full extent, not just the starting location, and ancillary info-level locations.

### CODE REVIEW

File: flycheck-relint.el [hunk 1]
```diff
@@ -76,14 +126,13 @@ CHECKER is this checker, and CALLBACK is the flycheck dispatch function."
...
 
-
 ;;;###autoload
 (defun flycheck-relint-setup ()
   "Setup flycheck-package.
 Add `emacs-lisp-relint' to `flycheck-checkers' and set up the checker chain."
   (interactive)
   (add-to-list 'flycheck-checkers 'emacs-lisp-relint t)
-  (flycheck-add-next-checker 'emacs-lisp-checkdoc 'emacs-lisp-relint t)
+  (flycheck-add-next-checker 'emacs-lisp 'emacs-lisp-relint t)
```
Comment (purcell) [inline flycheck-relint.el:135]: Why this change? Would this stop the `checkdoc` checker from running if the `relint` checker reports errors?

Comment (mattiase) [reply flycheck-relint.el:135]: I may have misunderstood how `flycheck-add-next-checker` works, but feared that relint wouldn't run unless checkdoc was run and didn't like that dependency – there is no particular relation between the two. (Like many others I don't use checkdoc myself as it's very opinionated and hardly ever finds actual problems.)

The way `flycheck-add-next-checker` is used here shouldn't add an error dependency, only sequencing, so relint is run after the elisp byte-compiler check.

Comment (mattiase) [reply flycheck-relint.el:135]: Maybe the CI failures are from `relint` (and `xr`) now requiring Emacs 27.

**Comment by purcell** [comment]
Okay, cool. Will merge as-is, then fix up the CI matrix. Thanks!
